src/image-copy/azext_imagecopy/custom.py

```diff
@@ -23,34 +25,57 @@ def imagecopy(source_resource_group_name, source_object_name, target_location,
 
     json_cmd_output = run_cli_command(cli_cmd, return_as_json=True)
-
-    source_os_disk_id = json_cmd_output['storageProfile']['osDisk']['managedDisk']['id']
+    source_os_disk_snapshot_url = ''
+    source_os_disk_snapshot_name = source_object_name + '_os_disk_snapshot'
     source_os_type = json_cmd_output['storageProfile']['osDisk']['osType']
-    logger.debug("source_os_disk_id: %s. source_os_type: %s",
-                 source_os_disk_id, source_os_type)
 
-    # create source snapshots
-    logger.warn("Creating source snapshot")
-    source_os_disk_snapshot_name = source_object_name + '_os_disk_snapshot'
-    cli_cmd = prepare_cli_command(['snapshot', 'create',
-                                   '--name', source_os_disk_snapshot_name,
-                                   '--resource-group', source_resource_group_name,
-                                   '--source', source_os_disk_id])
+    is_blob_disk = json_cmd_output['storageProfile']['osDisk']['managedDisk'] is None
 
-    run_cli_command(cli_cmd)
+    if not is_blob_disk:
+        source_os_disk_id = json_cmd_output['storageProfile']['osDisk']['managedDisk']['id']
+        logger.debug("source_os_disk_id: %s. source_os_type: %s", source_os_disk_id, source_os_type)
 
-    # Get SAS URL for the snapshotName
-    logger.warn("Getting sas url for the source snapshot")
-    cli_cmd = prepare_cli_command(['snapshot', 'grant-access',
-                                   '--name', source_os_disk_snapshot_name,
-                                   '--resource-group', source_resource_group_name,
-                                   '--duration-in-seconds', '3600'])
+        # create source snapshots
+        logger.warn("Creating source snapshot")
+        cli_cmd = prepare_cli_command(['snapshot', 'create',
+                                       '--name', source_os_disk_snapshot_name,
+                                       '--resource-group', source_resource_group_name,
+                                       '--source', source_os_disk_id])
 
-    json_output = run_cli_command(cli_cmd, return_as_json=True)
+        run_cli_command(cli_cmd)
+
+        # Get SAS URL for the snapshotName
+        logger.warn("Getting sas url for the source snapshot")
+        cli_cmd = prepare_cli_command(['snapshot', 'grant-access',
+                                       '--name', source_os_disk_snapshot_name,
+                                       '--resource-group', source_resource_group_name,
+                                       '--duration-in-seconds', '3600'])
 
-    source_os_disk_snapshot_url = json_output['accessSas']
-    logger.debug("source os disk snapshot url: %s",
-                 source_os_disk_snapshot_url)
+        json_output = run_cli_command(cli_cmd, return_as_json=True)
+
+        source_os_disk_snapshot_url = json_output['accessSas']
+        logger.debug("source os disk snapshot url: %s", source_os_disk_snapshot_url)
+    else:
+        logger.warn("blob OS disk detected")
```

Comment:
logger.warn writes to the console, so I think you should change that and let the user know what is the action performed, something like: Getting sas url for the source blob.